examples/hello-synchronization/src/main.rs

```diff
@@ -0,0 +1,205 @@
+const ARR_SIZE: usize = 128;
+
+async fn run() {
+    let mut local_patient_workgroup_results = [0_u32; ARR_SIZE];
+    let mut local_hasty_workgroup_results = local_patient_workgroup_results;
+
+    let instance = wgpu::Instance::default();
+    let adapter = instance
+        .request_adapter(&wgpu::RequestAdapterOptions::default())
+        .await
+        .unwrap();
+    let (device, queue) = adapter
+        .request_device(
+            &wgpu::DeviceDescriptor {
+                label: None,
+                features: wgpu::Features::empty(),
+                limits: wgpu::Limits::downlevel_defaults(),
+            },
+            None,
+        )
+        .await
+        .unwrap();
+
+    execute(
+        &device,
+        &queue,
+        &mut local_patient_workgroup_results[..],
+        &mut local_hasty_workgroup_results[..],
+    )
+    .await;
+
+    // Print data
+    log::info!("Patient results: {local_patient_workgroup_results:?}");
+    if !local_patient_workgroup_results.iter().any(|e| *e != 16) {
+        log::info!("patient_main was patient.");
+    } else {
+        log::error!("patient_main was not patient!");
+    }
+    log::info!("Hasty results: {local_hasty_workgroup_results:?}");
+    if local_hasty_workgroup_results.iter().any(|e| *e != 16) {
+        log::info!("hasty_main was not patient.");
+    } else {
+        log::info!("hasty_main got lucky.");
+    }
+}
+
+async fn execute(
+    device: &wgpu::Device,
+    queue: &wgpu::Queue,
+    local_patient_workgroup_results: &mut [u32],
+    local_hasty_workgroup_results: &mut [u32],
```

Comment:
Oh yeah. Kind of used to this thing from a C background and I've seen this before in Rust. I thought it was appropriate seeing as how we're all rather low level programmers around here with WGPU but if it's not to your liking, I can certainly change it. I can take in size params but what would be the best way to return new `Vec`'s? It might get a little confusing which is which with a tuple? Idk. I'll try that in the next push and see what you think.

Reply:
You can put them in a named struct that you return

Reply:
How does [30075a6](https://github.com/gfx-rs/wgpu/pull/3885/commits/30075a6f059bdae076b098e13c5a9af8b3a2ab08) look?

Reply:
Since I haven't gotten anything back, I'm going to assume that this is resolved.